component-test/test_dbus_proxy.py

```diff
@@ -486,6 +486,85 @@ def test_disallowed_iface_is_not_accessible(self,
         assert "My unique key" not in captured_stdout
 
 
+class TestProxyMethods(object):
+    """ This class tests methods configurations. There are two ways of setting methods
+        a string representing a method or an array of string representing couple of methods.
+    """
+
+    CONF_SINGLE_METHOD = """
+    {
+        "dbus-gateway-config-session": [{
+            "direction": "*",
+            "interface": "*",
+            "object-path": "*",
+            "method": "Method2"
+        }],
+        "dbus-gateway-config-system": []
+    }
+    """
+
+    CONF_ALL_METHODS = """
+    {
+        "dbus-gateway-config-session": [{
+            "direction": "*",
+            "interface": "*",
+            "object-path": "*",
+            "method": "*"
+        }],
+        "dbus-gateway-config-system": []
+    }
+    """
+
+    CONF_TWO_METHODS = """
+    {
+        "dbus-gateway-config-session": [{
+            "direction": "*",
+            "interface": "*",
+            "object-path": "*",
+            "method": ["Method1", "Method2"]
+        }],
+        "dbus-gateway-config-system": []
+    }
+    """
+
+    CONF_FOUR_METHODS = """
```

Comment:
Consider removing any unnecessary confs, e.g. the one that is the same but with additional strings in the method array.